homeassistant/components/select/__init__.py

```diff
@@ -99,6 +101,13 @@ def state(self) -> str | None:
     @property
     def options(self) -> list[str]:
         """Return a set of selectable options."""
+        if hasattr(self, "_attr_options"):
+            return self._attr_options
+        if (
+            hasattr(self, "entity_description")
+            and self.entity_description.options is not UNDEFINED
+        ):
+            return self.entity_description.options
         return self._attr_options
```

Comment:
This is now unreachable.

Reply:
Currently it a developper fails to override the property, and doesn't implement `_attr_options`, it raises `AttributeError`.

In this new scenario, I think we should still raise an error as fallback.

Reply:
Yes, we should raise if we don't implement things correctly. :+1: